models/base/scratch/default/snowplow_web_base_events_this_run.sql

```diff
@@ -6,313 +13,70 @@
-  where yauaa_context_dedupe_index = 1
-)
+{% if var('snowplow__enable_consent', false) -%}
+  {% do contexts.append({'name': var('snowplow__consent_cmp_visible'), 'prefix': 'cmp_visible', 'single_entity': True}) %}
+  {% do contexts.append({'name': var('snowplow__consent_preferences'), 'prefix': 'consent_pref', 'single_entity': True}) %}
 {% endif -%}
 
-select
-  e.*,
-  pc.page_view_id,
-
--- iab enrichment fields: set iab variable to true to enable
-  {{snowplow_web.get_iab_context_fields('iab')}},
+{% if var('snowplow__enable_cwv', false) -%}
+  {% do contexts.append({'name': var('snowplow__cwv_context'), 'prefix': 'cwv', 'single_entity': True}) %}
+{% endif -%}
 
-  -- ua parser enrichment fields
-  {{snowplow_web.get_ua_context_fields('ua')}},
+{% set base_events_query = snowplow_utils.base_create_snowplow_events_this_run(
+    sessions_this_run_table='snowplow_web_base_sessions_this_run',
+    session_identifiers=var('snowplow__session_identifiers', [{"table" : "events", "field" : "domain_sessionid"}]),
+    session_sql=var('snowplow__session_sql', none),
+    session_timestamp=var('snowplow__session_timestamp', 'collector_tstamp'),
+    derived_tstamp_partitioned=var('snowplow__derived_tstamp_partitioned', true),
+    days_late_allowed=var('snowplow__days_late_allowed', 3),
+    max_session_days=var('snowplow__max_session_days', 3),
+    app_ids=var('snowplow__app_ids', []),
+    snowplow_events_database=var('snowplow__database', target.database) if target.type not in ['databricks', 'spark'] else var('snowplow__databricks_catalog', 'hive_metastore') if target.type in ['databricks'] else var('snowplow__atomic_schema', 'atomic'),
+    snowplow_events_schema=var('snowplow__atomic_schema', 'atomic'),
+    snowplow_events_table=var('snowplow__events_table', 'events'),
+    entities_or_sdes=contexts) %}
+
+
+with base_query as (
+  {{ base_events_query }}
+)
 
-  -- yauaa enrichment fields
-  {{snowplow_web.get_yauaa_context_fields('ya')}}
-from events_this_run as e
-left join page_context_dedupe as pc
-  on e.event_id = pc.root_id and e.collector_tstamp = pc.root_tstamp
-{% if var('snowplow__enable_iab', false) -%}
-left join iab_context_dedupe as iab
-  on e.event_id = iab.iab_root_id and e.collector_tstamp = iab.iab_root_tstamp
-{%- endif %}
-{% if var('snowplow__enable_ua', false) -%}
-left join ua_context_dedupe as ua
-  on e.event_id = ua.ua_root_id and e.collector_tstamp = ua.ua_root_tstamp
-{%- endif %}
-{% if var('snowplow__enable_yauaa', false) -%}
-left join yauaa_context_dedupe as ya
-  on e.event_id = ya.yauaa_root_id and e.collector_tstamp = ya.yauaa_root_tstamp
-{%- endif %}
+{% set base_query_cols = get_column_schema_from_query( 'select * from (' + base_events_query +') a') %}
 
-where e.event_id_dedupe_index = 1
+select
+  {% for col in base_query_cols | map(attribute='name') | list -%}
+    {% if col == var("snowplow__page_view_context")~'_id' -%}
+      a.{{var("snowplow__page_view_context")}}_id as page_view_id
```

Comment:
This doesn't do anything, the page_view_context is prefixed with 'page_view' so the column exists regardless.

Reply:
Great point, I'll remove it!